src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -7417,6 +7417,14 @@ bool Compiler::fgAddrCouldBeNull(GenTree* addr)
             }
         }
     }
+#if defined(FEATURE_READYTORUN_COMPILER) && defined(TARGET_ARMARCH)
+    else if (addr->gtOper == GT_PHYSREG)
+    {
+        // Just to be conservative, ensure the srcReg is the known REG_R2R_INDIRECT_PARAM
+        // if we are claiming that addr cannot be null.
+        return addr->AsPhysReg()->gtSrcReg != REG_R2R_INDIRECT_PARAM;
```

Comment:
This doesn't seem safe to me. It's saying that r11 as a GT_PHYSREG can never be null. That might be true in the case where r11 is loaded as a r2r indirect special argument, but r11 can be used for other purposes, and this function might (eventually if not now) be used for other purposes as well, in which case this wouldn't always hold.

Reply:
Got it. In that case, do you think a better place would be inside `fgComputeLifeLIR()`?

Reply:
Yes, it feels like someplace inside `Removing dead call` code might be appropriate, but I'm not confident about that.

Reply:
Yes, I tried marking the node to be unused at place that you suggest (inside code where we iterate over the arguments of the call to see if they can be marked unused). but I hit assert going forward inside [lir](https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/lir.cpp#L1623). Instead what I was thinking was to add a case for `GT_IND` [here](https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/liveness.cpp#L2035) and the code will be exactly same as `default case` but with added condition that checks if the node is `GT_PHYSREG`  and if yes, remove it.

Reply:
Ok, made the fix in liveness.